### ISSUE TITLE
Feature request: Image posterization

### ISSUE THREAD

**Issue by trufanov-nok** [trufanov-nok/scantailor#31]
AAW: 

> ST*не постеризует картинки. От слова совсем. Если это так для всех версий ST, предлагаю записать в ToDo фичу: нарисовать на тексте зону, в свойствах которой можно указать цвет. При экспорте файлов данная зона меняет цвет подлежащего (чёрного) текста на указанный в свойствах зоны. И больше нигде ни на что не влияет. Этакий Djvu Pal. Мне кажется, что процентов 80 малоцветных книжек эта фича "облагородит". Сейчас ведь цветные заголовки глав отправляют в background, как обычные фотки. 
> ...
> получать нормальную пару сканов для FSD или DSM, особенно для случаев когда на скане И фотка И малоцветка. Пейзаж и крашеный колонтитул должны разойтись в background и foreground.

**Comment by ashipunov** [comment]
I agree. It was actually one of my ideas several years ago. Even primitive posterization will allow my img2djvu to run cpaldjvu on this particular image (img2djvu counts colors through ImageMagick).

**Comment by U235a** [comment]
Если есть 3-х цветное изображение (текст-черный, фон - белый и третий цет, в данном случае cyan),  то выделение третьего цвета можно легко осуществить с использованием метода главных компонент или PCA. Вектору в RGB пространстве, которому соответствует наибольшее собственное значение будет соответствовать направление черное-белое, а второму вектору - направление бесцветное-цветное. Архив с примером: https://yadi.sk/d/RqOziPyH3NJWUh
